src/js/components/DataTable/DataTable.js

```diff
@@ -273,6 +273,40 @@ const DataTable = ({
       ? { style: { minWidth: '100%' } }
       : undefined;
 
+  const pinnedColumns = columns
+    .map(
+      (pinnedColumn, index) =>
+        pinnedColumn.pin && { property: pinnedColumn.property, id: index },
+    )
+    .filter(n => n);
+
+  const getPinnedOffset = () => {
+    const pinnedPropertiesOrder = [];
+    const pinnedColumnsProperties = {};
+    const firstRowCells = headerRef?.current?.firstChild?.cells || [];
+
+    if (firstRowCells.length !== 0) {
+      pinnedColumns.forEach(({ property, id }, index) => {
```

Comment:
```suggestion
      pinnedColumns.forEach(({ property }, index) => {
```
Think we don't need `id` and can just use `index` in this function.